include/pcmki/pcmki_sched_allocate.h

```diff
@@ -30,6 +30,26 @@ struct resource_alloc_functions_s {
     void (*rsc_colocation_rh) (pe_resource_t *, pe_resource_t *,
                                pcmk__colocation_t *, pe_working_set_t *);
 
+    /*!
+     * \internal
+     * \brief Create list of all resources in colocations with a given resource
+     *
+     * Given a resource, create a list of all resources involved in mandatory
+     * colocations with it, whether directly or indirectly via chained colocations.
+     *
+     * \param[in] rsc             Resource to add to colocated list
+     * \param[in] orig_rsc        Resource originally requested
+     * \param[in] colocated_rscs  Existing list
+     *
+     * \return List of given resource and all resources involved in colocations
+     *
+     * \note This function is recursive; top-level callers should pass NULL as
+     *       \p colocated_rscs and \p orig_rsc, and the desired resource as
+     *       \p rsc. The recursive calls will use other values.
+     */
```

Comment:
Yep I wanted to avoid that originally too :) but this is a function pointer in a struct, with multiple implementations. We could wrap it to avoid calling it directly, but the function pointer itself would still need the recursive signature:

    static inline GList *
    pcmk__something(pe_resource_t *rsc)
    {
        return rsc->cmds->colocated_resources(rsc, NULL, NULL);
    }